lib/mongoid/indexable/specification.rb

```diff
@@ -63,39 +63,73 @@ def name
 
       private
 
-      # Normalize the spec, in case aliased fields are provided.
+      # Normalize the spec in place, in case aliased fields are provided.
```

Comment:
See https://github.com/mongodb/mongoid/pull/5267#discussion_r958781671. Ultimately nothing "has to be done in place", it's certainly possible to change the code so that returns and assignment is used everywhere, but it's gonna be 3x the lines of code, less readable, and more brittle IMHO.